core/mutex.c

```diff
@@ -131,3 +147,31 @@ void mutex_unlock_and_sleep(mutex_t *mutex)
     irq_restore(irqstate);
     thread_yield_higher();
 }
+
+#if IS_USED(MODULE_CORE_MUTEX_CANCEL)
+void mutex_cancel(mutex_t *mutex, thread_t *thread)
+{
+    unsigned irq_state = irq_disable();
+    if (thread->status >= STATUS_ON_RUNQUEUE) {
+        thread->wait_data = mutex;
+        irq_restore(irq_state);
+        return;
+    }
+
+    if ((mutex->queue.next != MUTEX_LOCKED) &&
+            (mutex->queue.next != NULL) &&
+            list_remove(&mutex->queue, (list_node_t *)&thread->rq_entry)) {
```

Comment:
```suggestion
    if ((mutex->queue.next != MUTEX_LOCKED)
        && (mutex->queue.next != NULL)
        && list_remove(&mutex->queue, (list_node_t *)&thread->rq_entry)) {
```

looks a bit more tidy, but that's just my personal taste